src/common/decorators/swagger.ts

```diff
@@ -126,11 +126,15 @@ type SupportedHttpStatusCodes =
  * @param messages - key value map of HTTP response code to their description that will be passed to appropriate `MethodDecorator`
  */
 export function ApiTransactionFailedResponse(
-  messages: Partial<Record<SupportedHttpStatusCodes, string>>
+  messages: Partial<Record<SupportedHttpStatusCodes, string | string[]>>
 ): ReturnType<typeof applyDecorators> {
   const decorators: MethodDecorator[] = [];
 
-  Object.entries(messages).forEach(([statusCode, description]) => {
+  Object.entries(messages).forEach(([statusCode, rawDescription]) => {
+    const description = Array.isArray(rawDescription)
+      ? '<ul>' + rawDescription.map(v => `<li>${v}</li>`).join(' ') + '</ul>'
```

Comment:
(style) Usually we don't do single letter variables, even if they are close to the definition.
```suggestion
      ? '<ul>' + rawDescription.map(description => `<li>${description}</li>`).join(' ') + '</ul>'
```

I still find myself doing single letters for things like this from time to time though.